core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUserHook.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import java.util.List;
+
+/**
+ * Interface for authorization User and Group hooks operation of the underlying access control
+ * system.
+ */
+public interface AuthorizationUserHook {
+  /**
+   * Add a new User to underlying access control system.
+   *
+   * @param user The user entity.
+   * @return True if the add User was successfully added, false if the add User was failed.
+   * @throws RuntimeException If adding the User encounters storage issues.
+   */
+  Boolean onAddUser(User user) throws RuntimeException;
+
+  /**
+   * Removes a User from underlying access control system.
+   *
+   * @param user The name of the User.
+   * @return True if the User was successfully removed, false if the remove User was failed.
+   * @throws RuntimeException If removing the User encounters storage issues.
+   */
+  Boolean onRemoveUser(String user) throws RuntimeException;
+
+  /**
+   * Gets a User from underlying access control system. Because User information already storage in
+   * the Gravition, so we don't need to get the User from the underlying access control system. We
+   * only need to check if the User exists in the underlying access control system.
+   *
+   * @param user The name of the User.
+   * @return The getting User instance.
+   * @throws RuntimeException If getting the User encounters storage issues.
+   */
+  User onGetUser(String user) throws RuntimeException;
```

Comment:
Also, is it required, how would we get a user from underlying system?

Reply:
Changed to `Boolean onCheckUser()...`